client/web/src/enterprise/insights/components/insights-view-grid/components/backend-insight/BackendInsight.tsx

```
@@ -82,9 +82,12 @@ export const BackendInsightView = forwardRef<HTMLElement, BackendInsightProps>((
                 },
             },
             onCompleted: data => {
-                const parsedData = createBackendInsightData({ ...insight, filters }, data.insightViews.nodes[0])
-                seriesToggleState.setSelectedSeriesIds([])
-                setInsightData(parsedData)
+                const node = data.insightViews.nodes[0]
+                if (node !== null) {
```

@chwarwick should we throw an error here if we didn't find insight here? Or this wouldn't be the case here because we filter out the null insights at the query before (where we fetch insight configurations)? 

Thought: Even if we filter out the null insights at the beginning it still should be possible to fetch insight configuration without a problem but encounter a problem and get a null value here in the backend insight card. Therefore I think we should handle error case here as well.

---

Also, I noticed that in the standalone insight page gql handler, we stopped polling manually in case of insight null value. Should we also handle this in this component?

---

At first I was thinking that if we have a null insight it would always have an error, which is true for our current implementation but the schema does allow that to be null and have no error so I should handle that.

The polling I don't think is an issue because it will get stoped by `isFetchingHistoricalData` being undefined

---

@vovakulikov  after looking at this more I do not believe there needs to be any additional changes. 

If there was an error in 1 more of the resolvers the insight is returned `null` and the apollo error is populated. This display the error in the card.

Polling will be stoped because a null/undefined insight will produce a falsey `isFetchingHistoricalData`